src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -132,29 +135,57 @@
+     * @param file              resulting merge filename (without .zip)
+     * @param mergeType         the type of merge to perform {@link MergeFeedsType}
+     * @param storeNewVersion   whether to store merged feed as new version
      */
     public MergeFeedsJob(Auth0UserProfile owner, Set<FeedVersion> feedVersions, String file,
-                         MergeFeedsType mergeType) {
+                         MergeFeedsType mergeType, boolean storeNewVersion) {
         super(owner, mergeType.equals(REGIONAL) ? "Merging project feeds" : "Merging feed versions",
             JobType.MERGE_FEED_VERSIONS);
         this.feedVersions = feedVersions;
-        // Grab parent feed source if performing non-regional merge (each version should share the
-        // same feed source).
-        this.feedSource =
-            mergeType.equals(REGIONAL) ? null : feedVersions.iterator().next().parentFeedSource();
         // Construct full filename with extension
         this.filename = String.format("%s.zip", file);
         // If the merge type is regional, the file string should be equivalent to projectId, which
         // is used by the client to download the merged feed upon job completion.
         this.projectId = mergeType.equals(REGIONAL) ? file : null;
         this.mergeType = mergeType;
         // Assuming job is successful, mergedVersion will contain the resulting feed version.
-        this.mergedVersion = mergeType.equals(REGIONAL) ? null : new FeedVersion(this.feedSource);
+        Project project = Persistence.projects.getById(projectId);
+        // Grab parent feed source depending on merge type.
+        FeedSource regionalFeedSource = null;
+        // If storing a regional merge as a new version, find the feed source designated by the project.
+        if (mergeType.equals(REGIONAL) && storeNewVersion) {
+            regionalFeedSource = Persistence.feedSources.getById(project.regionalFeedSourceId);
+            // Create new feed source if this is the first regional merge.
+            if (regionalFeedSource == null) {
+                regionalFeedSource = new FeedSource("REGIONAL MERGE", project.id, REGIONAL_MERGE);
+                // Store new feed source.
+                Persistence.feedSources.create(regionalFeedSource);
+                // Update regional feed source ID on project.
+                project.regionalFeedSourceId = regionalFeedSource.id;
+                Persistence.projects.replace(project.id, project);
+            }
+        }
+        // Assign regional feed source or simply the first parent feed source found in the feed version list (these
+        // should all belong to the same feed source if the merge is not regional).
+        this.feedSource = mergeType.equals(REGIONAL)
+            ? regionalFeedSource
+            : feedVersions.iterator().next().parentFeedSource();
+        // Merged version will be null if the new version should not be stored.
```

Comment:
Can you also write some JavaDoc about how this class variable can be null up in the variable definition?